src/adapter/src/coord.rs

```diff
@@ -3048,10 +3051,20 @@ pub async fn load_remote_system_parameters(
                 })
                 .collect();
             tracing::info!("parameter sync on boot: end sync");
-            return Ok(Some(ops));
+            Ok(Some(ops))
+        };
+        return if !storage.has_system_config_synced_once().await? {
```

Comment:
nit: I find the `return if` construction here a touch unexpected. It's certainly readable enough, so very much feel free to ignore this, but I think the more usual expression here would be to have the `if/else` as the last expression in the fn and for each branch in it to evaluate to a `Result<Option<BTreeMap<String, OwnedVarInput>>, AdapterError>`

Reply:
I updated to remove the `return if`. We need to nest the last statement in an `else`, but I don't think that's a big deal.